src/plugins/kibana/public/visualize/wizard/wizard.js

```diff
@@ -56,16 +56,8 @@ module.controller('VisualizeWizardStep2', function ($route, $scope, $location, t
     list: $route.current.locals.indexPatternIds
   };
 
-  $scope.$watch('stepTwoMode', function (mode) {
-    if (mode === 'new') {
-      if ($scope.indexPattern.list && $scope.indexPattern.list.length === 1) {
-        $scope.indexPattern.selection = $scope.indexPattern.list[0];
-      }
-    }
-  });
-
-  $scope.$watch('indexPattern.selection', function (pattern) {
+  $scope.goToUrl = function (pattern) {
```

Comment:
I don't think this is required, the index pattern selector is a drop down in the design. Though for future reference, for stuff like this you can use kbn-href